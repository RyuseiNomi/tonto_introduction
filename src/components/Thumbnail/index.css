/* ========= スマホサイズ_1 =========== */
@media (min-width: 280px) {
  picture source {
    padding: 30px 0 10px 0;
    max-width: 180px;
    max-height: 180px
  }

  picture img {
    padding: 30px 0 10px 0;
    max-width: 180px;
    max-height: 180px
  }
}

/* ========= スマホサイズ_2 =========== */
@media (min-width: 576px) {
  picture source {
    padding: 30px 0 10px 0;
    max-width: 240px;
    max-height: 240px
  }

  picture img {
    padding: 30px 0 10px 0;
    max-width: 240px;
    max-height: 240px
  }
}

/* ========= タブレットサイズ =========== */
@media (min-width: 768px) {
  picture {
    display: flex;
    justify-content: right;
  }

  picture source {
    padding: 50px 10px 50px 0;
    max-width: 280px;
    max-height: 280px
  }

  picture img {
    padding: 50px 10px 50px 0;
    max-width: 280px;
    max-height: 280px
  }
}

/* ========= PCサイズ =========== */
@media (min-width: 992px) {
  picture {
    display: flex;
    justify-content: right;
  }

  picture source {
    padding: 50px 20px 50px 0;
    max-width: 320px;
    max-height: 320px
  }

  picture img {
    padding: 50px 20px 50px 0;
    max-width: 320px;
    max-height: 320px
  }
}

/* ========= PCサイズ_2 =========== */
@media (min-width: 1120px) {
  picture {
    display: flex;
    justify-content: right;
  }

  picture source {
    padding: 50px 30px 50px 0;
    max-width: 320px;
    max-height: 320px
  }

  picture img {
    padding: 50px 30px 50px 0;
    max-width: 320px;
    max-height: 320px
  }
}
