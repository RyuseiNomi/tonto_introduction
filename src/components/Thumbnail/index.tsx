import React from 'react';
import './index.css';
import { Row } from 'antd';  
import { TwitterOutlined } from '@ant-design/icons';

type Props = {
  name: {
    jp: string;
    en: string;
  }
  twitter_url: string;
}

function Thumbnail(props: Props) {

  const { name, twitter_url } = props;

  return (
    <div className="thumbnail-image">
      <picture>
        <source type="image/webp" srcSet={`${process.env.PUBLIC_URL}/thumbnail.webp`} ></source>
        <img src={`${process.env.PUBLIC_URL}/thumbnail.jpeg`} alt="thumbnail" />
      </picture>
      <Row justify="center">
        <p>{name.jp}</p>
        <a href={twitter_url} rel="noopener noreferrer" target="_blank"><TwitterOutlined /></a>
      </Row>
    </div>
  );
}

export default Thumbnail;
