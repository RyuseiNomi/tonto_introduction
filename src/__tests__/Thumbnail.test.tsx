import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import Thumbnail from './../components/Thumbnail';

let container = null;

beforeEach(() => {
  // conteiner の定義
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // 定義した container の除去
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

// =============== snapshot test =================
it('サムネイル画像が表示されていること', () => {
  const component = renderer.create(<Thumbnail />)
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

// ================ act tests ====================
it('サムネイル画像が img 要素で表示されていること', () => {
  act(() => {
    render(<Thumbnail />, container);
  });
  const displayedImage = document.querySelector("img");
  expect(displayedImage.src).toContain("thumbnail");
});

it('サムネイル画像が source 要素で表示されていること', () => {
  act(() => {
    render(<Thumbnail />, container);
  });
  const displayedImage = document.querySelector("source");
  expect(displayedImage.srcset).toBe("/thumbnail.webp");
});
