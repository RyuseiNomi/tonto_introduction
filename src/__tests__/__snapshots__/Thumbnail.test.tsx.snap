// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`サムネイル画像が表示されていること 1`] = `
<div
  aria-label="thumbnail-wrapper"
  className="thumbnail-image"
>
  <picture>
    <source
      srcSet="/thumbnail.webp"
      type="image/webp"
    />
    <img
      alt="thumbnail"
      src="/thumbnail.jpeg"
    />
  </picture>
</div>
`;
